Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -479,19 +480,22 @@ bool FullscreenManager::willEnterFullscreen(Element& element, HTMLMediaElementEn
 
     if (backForwardCacheState() != Document::NotInBackForwardCache) {
         ERROR_LOG(LOGIDENTIFIER, "Document in the BackForwardCache; bailing");
+        resolvePendingPromise();
```

Comment:
Should this not be `rejectPendingPromise()`?

I see references to `m_pendingPromise->reject(Exception { ExceptionCode::TypeError });` removed; does this mean we never reject the fullscreen promise any longer?

Reply:
> Should this not be rejectPendingPromise()?

Good catch! There might be a reason I decided to go with resolve (maybe tests?), but I guess I'll find out.

> I see references to m_pendingPromise->reject(Exception { ExceptionCode::TypeError }); removed; does this mean we never reject the fullscreen promise any longer?

I refactored those in https://github.com/WebKit/WebKit/commit/bb6320734c7fe75a57c2fd7abdc2673266ac19d1 , they're mostly happening from one method now